dolphin/api/api_utils.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2010 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import webob
+from oslo_log import log
+
+LOG = log.getLogger(__name__)
+
+
+def remove_invalid_options(context, search_options, allowed_search_options):
+    """Remove search options that are not valid for API/context."""
+    unknown_options = [opt for opt in search_options
+                       if opt not in allowed_search_options]
+    bad_options = ", ".join(unknown_options)
+    LOG.debug("Removing options '%(bad_options)s' from query",
+              {"bad_options": bad_options})
+    for opt in unknown_options:
+        del search_options[opt]
+
+
+def get_pagination_params(query_params):
+    """Return marker, limit, offset tuple from request."""
+    limit = query_params.get('limit', None)
+    offset = query_params.get('offset', None)
+    marker = query_params.get('marker', None)
```

Comment:
did you test here? why default value of them are None?

Reply:
added default max-limit and tested with filter parametrs like marker,offset,sort etc.